File: 1]. DSA/1]. Data Structures/01]. Array/C++/_038)_Find_SubArray_of_Given_Sum.cpp
#include<iostream>
using namespace std;

int subArraySum(int arr[], int n, int given_sum)
{
    for(int i=0; i<n; i++)
    {
        int sum = 0;
        for(int j=i; j<n; j++)
        {
            sum += arr[j];
            if(sum == given_sum)
            {
                cout<<"Sum found between indexes "<< i <<" and "<< j <<endl;
                return true;
            }
        }
    }
    cout<<"No subArray found"<<endl;
}

int main()
{
    int arr[] = {1, 4, 20, 3, 10, 5}, n=6, given_sum=23;
    subArraySum(arr, n, given_sum);
}

File: 1]. DSA + CP/1]. DSA/1]. Data Structures/01]. Array/Dart/_038)_Find_SubArray_of_Given_Sum.dart
void subArraySum(List<int> arr, int reqSum) {
  for (int i = 0; i < arr.length; i++) {
    int sum = 0;
    for (int j = i; j < arr.length; j++) {
      sum += arr[j];
      if (sum == reqSum) {
        print('Sum found between indexes $i to $j.');
        return;
      }
    }
  }
  print('Sum not found.');
}

void main() {
  var arr = [1, 4, 20, 3, 10, 5];
  int sum = 23;
  subArraySum(arr, sum);
}
